enhancements/installer/credentials-management-outside-openshift-cluster.md

```diff
@@ -0,0 +1,130 @@
+
+### Risks and Mitigations
+
+## Design Details
+
+We intend to build an optional tool **ccoctl** which will handle credentials management of the cluster in *manual* mode. 
+The following is the set of requirements for the current prototype (for AWS). We can have something similar for other 
+platforms.
+
+* ccoctl should be able to setup Identity Provider to authenticate OpenShift components
+* ccoctl should be able to take a list of CredentialsRequests from the release image and create/update 
+  Roles/Users/ServicePrincipals/ServiceAccounts with appropriate permissions.
+* ccoctl should be able to take list of CredentialsRequests for a release image, and the Identity Provider URL to generate 
+  the objects that need to be passed to the installer for successful installation
+* ccoctl should be able to delete all the resources that it had created in the cloud
+
+We envision **ccoctl** as a recommended tool to setup credentials on new platforms but customers are free to use other 
+tools like Terraform/AWS CloudFormation to do the above mentioned tasks. Read more about AWS implementation details [here](https://github.com/openshift/cloud-credential-operator/blob/master/docs/ccoctl.md)
+
+We also plan to have a detailed documentation in place to guide new cloud providers to implement *manual* mode.
+
+### Open Questions
+
+### Test Plan
+
+We plan to have a e2e test that will externally set up a credentials management infrastructure and then kickstart 
+install in a *manual* mode.
+
+### Graduation Criteria
+
+### Upgrade / Downgrade Strategy
+
+To upgrade we need to execute following steps:
+* Examine the CredentialsRequests in the new OpenShift release. Check if permissions in the existing CredentialsRequest
+  have changed.
+* Create/update credentials in the underlying cloud provider, and also create/update Kubernetes Secrets in the correct 
+  namespaces to satisfy all CredentialsRequests in the new release.
+* Set an appropriate annotation `cloudcredential.openshift.io/upgradeable-to` to a new upgradable version.
+
+We also plan to have provision in ccoctl to help in upgrades. Details in [this](https://issues.redhat.com/browse/CCO-84) card.
+
+### Version Skew Strategy
+
+## Implementation History
+
+We currently have a work-in-progress CLI tool [ccoctl](https://github.com/openshift/cloud-credential-operator/blob/master/docs/ccoctl.md) 
+to create and manage cloud credentials outside the cluster for AWS cloud. The design details of this tool is discussed above.
+
+## Drawbacks
+
+* Taking the credentials management outside the cluster will create additional overhead for the customer to make sure all
+  the required infrastructure is in place before starting the installation process. Current tooling we have only supports
+  the AWS cloud, we do not have anything planned for other cloud providers.
+* Push-button upgrades will not work in *manual* mode as the cluster no longer has the admin credentials to mint credentials 
```

Comment:
Is there a way that push-button upgrades could work for upgrades where there are no changes in CredentialsRequests?

Reply:
As designed today, no, you will always need to indicate that your cluster has been prepped for the next upgrade. We wanted users who opt into Manual mode to be familiar with this flow so it was decided this would be better for now.

Reply:
Are the "users who opt into Manual mode" the users with existing clusters?

Reply:
We would not change anything automatically on anyone, but yes I believe a current cluster admin could flip their cluster to manual mode if they took explicit action to do so.